src/metabase/models/query.clj

```diff
@@ -38,21 +38,23 @@
 (defn- update-rolling-average-execution-time!
   "Update the rolling average execution time for query with `query-hash`. Returns `true` if a record was updated,
    or `false` if no matching records were found."
-  ^Boolean [query, ^bytes query-hash, ^Integer execution-time-ms]
-  (let [avg-execution-time (hx/cast (int-casting-type) (hx/round (hx/+ (hx/* 0.9 :average_execution_time)
-                                                                       (*    0.1 execution-time-ms))
-                                                                 0))]
+  ^Boolean [query ^bytes query-hash ^Integer execution-time-ms]
+  (let [avg-execution-time (h2x/cast (int-casting-type) (h2x/round (h2x/+ (h2x/* [:inline 0.9] :average_execution_time)
```

Comment:
We don't strictly **need** to use `:inline` here, but it's easier to read the resulting query and more efficient too